src/normalize.rs

```diff
@@ -311,14 +320,15 @@ impl<'a> Filter<'a> {
                     }
                 }
             }
-            if other_crate && self.normalization >= WorkspaceLines {
+            if (other_crate || is_rust_src) && self.normalization >= WorkspaceLines {
                 // Blank out line numbers for this particular error since rustc
                 // tends to reach into code from outside of the test case. The
                 // test stderr shouldn't need to be updated every time we touch
                 // those files.
                 hide_trailing_numbers(&mut line);
-                self.hide_numbers = 1;
-                while let Some(next_line) = self.all_lines.get(index + self.hide_numbers) {
+                self.numbers_to_hide_are_rust_src = is_rust_src;
+                self.hide_numbers = 0;
+                while let Some(next_line) = self.all_lines.get(index + self.hide_numbers + 1) {
```

Comment:
It seems to me that there was an off-by-one error here causing `self.hide_numbers` to be one higher than it needed to be. That didn't matter while the operation was just trimming leading numbers, resulted in a noop in the last line, but removing whole lines was more destructive.